src/esm_runscripts/filedicts.py

```diff
@@ -73,6 +72,23 @@ def inner_method(self, *args, **kwargs):
     return inner_method
 
 
+def _fname_has_date_stamp_info(fname, date, reqs=["%Y", "%m", "%d"]):
+    date_attrs = {
+        "%Y": "syear",
+        "%m": "smonth",
+        "%d": "sday",
+        "%H": "shour",
+        "%M": "sminute",
+        "%S": "ssecond",
+    }
+    required_attrs = [getattr(date, v) for k, v in date_attrs.items() if k in reqs]
+    # all(attr in fname for attr in required_attrs)
+    for attr in required_attrs:
```

Comment:
I think this attributes will be resolved by the time we call filedicts in the recipe. Not resolving variables before that will likely generate all sorts of problems. So I guess this function will not do anything in a real word scenario?

Reply:
Probably not; I pulled this one out because putting it into the other function was making it longer than I wanted.

Reply:
I should note: the overall goal of the functionality was: don't add a "redundant" date stamp if you can already uniquely identify the date of the file from the filename. Probably this never happened in the old version, so probably this one was overkill.

Reply:
I'm fine with including it anyway. We might need it later on.

Reply:
There is still a docstring missing though, which I will add. Also I am sure this function will misbehave with files that happen to have `01` or something in their name anyway. Or if someone uses a crazy `expid`...but let's just be optimistic and assume that will not happen.